data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java

```diff
@@ -300,7 +301,36 @@ public SafeFuture<Void> prepareBeaconProposer(
 
   public SafeFuture<Void> registerValidators(
       SszList<SignedValidatorRegistration> validatorRegistrations) {
-    return validatorApiChannel.registerValidators(validatorRegistrations);
+    return validatorApiChannel
```

Comment:
I have 2 doubts here:

1- putting filtering here means that it will only be applied when registering from a remote  VC, and it will not be applied in single process.
I see this ValidatorDataProvider as a very thin layer. We should not have business logic unless we really want to differentiate between in process and remote VC handling.

2- if we filter upfront it means that validators in unknown status will not go in our BN data structure. Which means that, if VC doesn't keep calling the registration endpoint, the validator could become known at some point in future. It is an edge case but in theory we loose this info.

Reply:
sorry for `2-`. It is totally irrelevant considering what we are trying to achieve with this PR :)